File: src/main/java/Airlines/AirlineTicket/Seat/Models/SeatService.java
package Airlines.AirlineTicket.Seat.Models;

import Airlines.AirlineTicket.Seat.Seat;


import java.util.List;


public interface SeatService {

    Seat saveSeat(Seat seat);

    List<Seat>getAllSeats();

    Seat getSeatById(Long id);

    Seat updateSeat (Long id , Seat seatDetails);

    void deleteSeat(Long id);




}

File: frontend/src/services/SeatService.js
import axios from 'axios';

const SEAT_API_BASE_URL = "http://localhost:8080/seats";

class SeatService {
    getAllSeats() {
        return axios.get(SEAT_API_BASE_URL);
    }

    getSeatById(id) {
        return axios.get(`${SEAT_API_BASE_URL}/${id}`);
    }

    createSeat(seat) {
        return axios.post(SEAT_API_BASE_URL, seat);
    }

    updateSeat(id, seat) {
        return axios.put(`${SEAT_API_BASE_URL}/${id}`, seat);
    }

    deleteSeat(id) {
        return axios.delete(`${SEAT_API_BASE_URL}/${id}`);
    }
}

export default new SeatService();
